packages/@react-spectrum/menu/docs/MenuTrigger.mdx

```diff
@@ -0,0 +1,416 @@
+<!-- Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License. -->
+
+import {Layout} from '@react-spectrum/docs';
+export default Layout;
+
+import docs from 'docs:@react-spectrum/menu';
+import {HeaderInfo, PropTable} from '@react-spectrum/docs';
+import packageData from '../package.json';
+
+```jsx import
+import {ActionButton} from '@react-spectrum/button'
+import Copy from '@spectrum-icons/workflow/Copy';
+import Cut from '@spectrum-icons/workflow/Cut';
+import {Item, Menu, MenuTrigger, Section} from '@react-spectrum/menu';
+import {Keyboard, Text} from '@react-spectrum/typography';
+import Paste from '@spectrum-icons/workflow/Paste';
+```
+
+# MenuTrigger
+
+<p>{docs.exports.MenuTrigger.description}</p>
+
+<HeaderInfo
+  packageData={packageData}
+  componentNames={['MenuTrigger']}
+  sourceData={[
+    {type: 'Spectrum', url: 'https://spectrum.adobe.com/page/popovers/'}
+  ]} />
+
+## Example
+
+```tsx example
+<MenuTrigger>
+  <ActionButton>
+      Menu Button
+  </ActionButton>
+  <Menu items={[{name: 'Aardvark'}, {name: 'Kangaroo'}, {name: 'Snake'}]} itemKey="name">
+    {item => <Item>{item.name}</Item>}
+  </Menu>
+</MenuTrigger>
+```
+
+## Content
+
+The MenuTrigger accepts exactly two children: the Menu and the element which
+triggers the opening of the Menu. The trigger must be the first child passed into
+the MenuTrigger and should be an element that supports press events.
+
+If the Menu is open it will close on blur or scroll events.
+
+The `defaultSelectedKeys` prop can be used to set the default state of the Menu
+(uncontrolled). Alternatively, the `selectedKeys` prop can be used to make the
+selected state of the Menu controlled.
+
+```tsx example
+<MenuTrigger>
+  <ActionButton>
+      Menu Button Controlled
+  </ActionButton>
+  <Menu items={[{name: 'Aardvark'}, {name: 'Kangaroo'}, {name: 'Snake'}]} itemKey="name" selectedKeys={['Kangaroo']}>
+    {item => <Item>{item.name}</Item>}
+  </Menu>
+</MenuTrigger>
+<MenuTrigger>
+  <ActionButton>
+      Menu Button Uncontrolled
+  </ActionButton>
+  <Menu items={[{name: 'Aardvark'}, {name: 'Kangaroo'}, {name: 'Snake'}]} itemKey="name" defaultSelectedKeys={['Kangaroo']}>
+    {item => <Item>{item.name}</Item>}
+  </Menu>
+</MenuTrigger>
+```
```

Comment:
I noticed that these examples close the Menu immediately after selecting a different key, probably would be more useful to put closeOnSelect to false or something so that the difference between the controlled and uncontrolled menus is more apparent

(Not too certain why pressing space on a menu item is still closing the menu though, thought it should toggle selection without closing the menu...)

Reply:
It might be a weird effect of selectionMode defaulting to `none` in menu.